packages/esm-system-admin-app/src/change-password-modal/change-password-modal.component.tsx

```diff
@@ -0,0 +1,203 @@
+        errMsg = t('atLeastOneUppercase', 'At least one Uppercase');
+      } else if (!lowercasePassword) {
+        errMsg = t('atLeastOneLowercase', 'At least one Lowercase');
+      } else if (!digitsPassword) {
+        errMsg = t('atLeastOneDigit', 'At least one digit');
+      } else if (!minLengthPassword) {
+        errMsg = t('minimum8Characters', 'Minimum 8 characters');
+      } else if (passwordInput.oldPassword.length > 0 && passwordInput.newPassword === passwordInput.oldPassword) {
+        errMsg = t('newPasswordMustNotBeTheSameAsOld', 'New password must not be the same as password');
+      } else {
+        errMsg = '';
+        setIsNewPasswordInvalid(false);
+      }
+      setNewPasswordErr(errMsg);
+    } else if (
+      passwordInputFieldName === 'confirmPassword' ||
+      (passwordInputFieldName === 'newPassword' && passwordInput.confirmPassword.length > 0)
+    ) {
+      if (passwordInput.confirmPassword !== passwordInput.newPassword) {
+        setConfirmPasswordError(t('confirmPasswordMustBeTheSameAsNew', 'Confirm password must be the same as new'));
+      } else {
+        setConfirmPasswordError('');
+        setIsConfirmPasswordInvalid(false);
+      }
+    } else {
+      if (passwordInput.newPassword.length > 0 && passwordInput.newPassword === passwordInput.oldPassword) {
+        setOldPasswordErr(t('oldPasswordMustNotBeTheSameAsNew', 'Old password must not be the same as new'));
+      } else {
+        setOldPasswordErr('');
+        setIsOldPasswordInvalid(false);
+      }
+    }
+  };
+
+  const handleSubmit = useCallback(
+    async (evt: React.FormEvent<HTMLFormElement>) => {
+      evt.preventDefault();
+      evt.stopPropagation();
+
+      try {
+        setIsSavingPassword(true);
+        await performPasswordChange(passwordInput.oldPassword, passwordInput.confirmPassword).then(() => {
+          close();
+          navigate({ to: `\${openmrsSpaBase}/logout` });
+          showToast({
+            title: t('userPassword', 'User password'),
+            description: t('userPasswordUpdated', 'User password updated successfully'),
+            kind: 'success',
+          });
+        });
+      } catch (error) {
+        setIsSavingPassword(false);
+        setErrorMessage(`${t('invalidPasswordCredentials', 'Invalid password provided')}: ${error.message}`);
+      }
+
+      return false;
+    },
+    [close, passwordInput.confirmPassword, passwordInput.oldPassword, t],
+  );
+
+  return (
+    <>
+      <ModalHeader closeModal={close} title={t('changePassword', 'Change Password')} />
+      <ModalBody>
+        <div className={styles['input-group']}>
+          {errorMessage && (
+            <InlineNotification
+              className={styles.errorMessage}
+              kind="error"
+              /**
+               * This comment tells i18n to still keep the following translation keys (used as value for: errorMessage):
+               * t('invalidPasswordCredentials')
```

Comment:
Do we need this comment? It's not dynamically constructed. I think, if the key isn't picked up by default, we can just use standard string concatenation.